File: 2026POTD/JANUARY/06-01-2026/1161.go
func maxLevelSum(root *TreeNode) int {
    q := []*TreeNode{root}
    level, ans := 1, 1
    maxSum := -1 << 60

    for len(q) > 0 {
        size := len(q)
        sum := 0

        for i := 0; i < size; i++ {
            node := q[0]
            q = q[1:]
            sum += node.Val
            if node.Left != nil {
                q = append(q, node.Left)
            }
            if node.Right != nil {
                q = append(q, node.Right)
            }
        }

        if sum > maxSum {
            maxSum = sum
            ans = level
        }
        level++
    }
    return ans
}

File: 2026POTD/JANUARY/06-01-2026/1161.swift
class Solution {
    func maxLevelSum(_ root: TreeNode?) -> Int {
        var q: [TreeNode] = [root!]
        var level = 1, ans = 1
        var maxSum = Int.min

        while !q.isEmpty {
            let size = q.count
            var sum = 0

            for _ in 0..<size {
                let node = q.removeFirst()
                sum += node.val
                if let left = node.left { q.append(left) }
                if let right = node.right { q.append(right) }
            }

            if sum > maxSum {
                maxSum = sum
                ans = level
            }
            level += 1
        }
        return ans
    }
}